lib/spack/spack/test/conftest.py

```diff
@@ -40,6 +40,14 @@
 from spack.version import Version
 
 
+@pytest.fixture
+def no_path_access(monkeypatch):
```

Comment:
Although I read your point at https://github.com/spack/spack/pull/12733#discussion_r327760429, my general preference is still to avoid monkeypatching system functions whenever possible. For example could `stage.create_stage_root` take an extra parameter that is an object which can determine the accessibility of file paths? In that case `test_create_stage_root` could pass in a mocked version of that object.

Reply:
It's code so it can be done; however, I thought there was a project convention of **not** adding options just to make testing easier.

Are there actual use cases you can think of for such an option?

Reply:
Reworded in previous commit.

Reply:
> It's code so it can be done; however, I thought there was a project convention of not adding options just to make testing easier.

With my suggestion, the parameter would always be used: in the non-test case it would take an object/function that actually calls `os.path.exists` or the like, and in the test case it would take this mocked object.

Reply:
Can this be a follow-on issue given the pending release?

Reply:
yes

Reply:
Great.  Created issue https://github.com/spack/spack/issues/13240.